packages/api/src/middleware/validation/checkAnimalOrBatch.js

```diff
@@ -607,18 +614,24 @@ export function checkRemoveAnimalOrBatch(animalOrBatchKey) {
  * );
  *
  */
-export function checkDeleteAnimalOrBatch(animalOrBatchKey) {
+export function checkDeleteAnimalOrBatch(animalOrBatchKey, dryRun = false) {
```

Comment:
https://dx.apiture.com/docs/api/concepts/dry-runs/#:~:text=Performing%20a%20Dry%20Run,error%20data%20and%20response%20code.

https://stackoverflow.com/questions/31967684/dry-run-strategy-for-rest-api

https://www.google.com/search?q=dry+run+meaning&oq=dry+run+&gs_lcrp=EgZjaHJvbWUqCggBEAAYsQMYgAQyBggAEEUYOTIKCAEQABixAxiABDIHCAIQABiABDIHCAMQABiABDIHCAQQABiABDIHCAUQABiABDIHCAYQABiABDIHCAcQABiABDIHCAgQABiABDIHCAkQABiABNIBCDI5MDdqMGo3qAIAsAIA&sourceid=chrome&ie=UTF-8

Reply:
I understand, but the strategy described in the links you posted is not what's being implemented here. In the links, what's being described is a situation where the client has the intention of doing a certain operation and needs to validate if that's possible -- translated to our situation here, an actual dry run would happen if we called this endpoint **after** the user selected the "Created in error" option. If that were the case, then yes, the strategy would be what's described in the links, we'd just be validating if we can delete before actually deleting. Here, on the other hand, we're checking if the operation is possible but we have no idea if we'll actually delete or not afterwards. It's for that same reason that using the "delete" HTTP method is wrong -- it's not the same as if we tried to delete and got an error, because when we call this endpoint we have no intention of deleting the resource yet.